src/Modal.js

```diff
@@ -122,7 +122,8 @@ class Modal extends React.Component {
       this._element = null;
     }
 
-    const classes = document.body.className.replace('modal-open', '');
+    // Use regex to prevent matching `modal-open` as part of a different class, e.g. `my-modal-opened`
+    const classes = document.body.className.replace(/(^| )modal-open( |$)/, '');
```

Comment:
Great catch!
This was adapted from [Element.classList polyfill](https://developer.mozilla.org/en/docs/Web/API/Element/classList#Polyfill).
I guess I could also replace with `' '` instead of `''`. 
If you would opt for using `\b`, I would at least follow the `.replace(/\bmodal-open\b/, '')` with `.replace(/\s{2,}/g, ' ')`, i.e. replace all occurrences of 2 or more spaces with a single space. Otherwise, it looks weird and could lead to a broken test case if someone would expect single space and would get 2 spaces.

I would also suggest using `document.body.classList` in the future. Bootstrap 4 supports only IE10+ and classList is already supported in IE10. With `classList.remove`, there would be no such problem in the first place 😄